src/main/java/com/codingame/core/GameManager.java

```diff
@@ -0,0 +1,60 @@
+package com.codingame.core;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Singleton;
+
+@Singleton
+public final class GameManager<T extends AbstractPlayer> {
+    private List<T> players;
+    private int turnMaxTime = 50;
+    private int maxTurn = 200;
+
+    public List<T> getPlayers() {
+        return players;
+    }
+
+    public T getPlayer(int index) {
+        return players.get(index);
+    }
+
+    public List<T> getActivePlayers() {
+        List<T> actives = new ArrayList<T>();
+        for (T player : players) {
+            if (player.isActive()) {
+                actives.add(player);
+            }
+        }
+
+        return actives;
```

Comment:
Because you choose java 8 implementation, you can use streams. Something like:

```java
return players.stream()
    .filter(AbstractPlayer::isActive)
    .collect(Collectors.toList());
```

Reply:
I tried, but I did not find how to return a toList of generics

Reply:
done in next PR